pkg/appmanager/resourceConfig.go

```diff
@@ -1440,6 +1774,16 @@ func (appMgr *Manager) handleRouteTls(
 				rc.Virtual.AddIRule(abPathIRuleName)
 			} else {
 				rc.AddRuleToPolicy(policyName, rule)
+				if len(appRootRules) == 2 {
+					rc.AddRuleToPolicy(policyName, appRootRules[0])
+					appMgr.AddRuleToAnnotationsMap(virtualName, poolName, appRootRules[0].Name)
+					rc.AddRuleToPolicy(policyName, appRootRules[1])
+					appMgr.AddRuleToAnnotationsMap(virtualName, poolName, appRootRules[1].Name)
+				}
+				if urlRewriteRule != nil {
+					rc.AddRuleToPolicy(policyName, urlRewriteRule)
+					appMgr.AddRuleToAnnotationsMap(virtualName, poolName, urlRewriteRule.Name)
+				}
```

Comment:
This block looks like it gets reused 4 times. Can we move it into its own small function?

Reply:
moved